### PR TITLE
Ancillary and smoothing coefficient improvements

### PR DESCRIPTION
**Pull request by bayliffe**
This PR does a few things:

- Finishes some tidy up of IMPROVER ancillary files that was previously agreed. This consolidates all ancillary creation code under the generate_ancillaries directory; previously some was here and some was under utilities/generate_ancillary.py.
- Removes the ability to specify a coefficient when generating orographic smoothing coefficients. The resulting values are scaled between and min and max values which makes the use of the coefficient ineffectual / pointless.
- Modified the metadata for the smoothing coefficient cubes to include the power that has been used to generate them so that this is recorded; previously we have only used a power of 1 so this was not needed, but we are now going to start using other powers.
- Added an acceptance test for the generate_orographic_smoothing_coefficients CLI that includes an alternative power. The existing acceptance test data for all the other orographic smoothing coefficients tests has changed **only** to now include the power metadata.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py  10
- improver/generate_ancillaries/generate_orographic_smoothing_coefficients.py  9
- improver/generate_ancillaries/generate_svp_table.py  5
- improver_tests/generate_ancillaries/test_SaturatedVapourPressureTable.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6421563&bid=20804685)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/generate_ancillaries/test_SaturatedVapourPressureTable.py  1
- improver/generate_ancillaries/generate_svp_table.py  5
- improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py  10
- improver/generate_ancillaries/generate_orographic_smoothing_coefficients.py  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6421563&bid=20804685)

File: improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py [hunk 1]
```diff
@@ -154,21 +155,21 @@ def test_scale_smoothing_coefficients(smoothing_coefficients):
...
     assert_array_almost_equal(result[0, :], expected)
 
-    # Coefficient = 0.5, power = 0.5
-    plugin = OrographicSmoothingCoefficients(coefficient=0.5, power=0.5)
-    expected = np.array([0.0, 0.353553, 0.5, 1.118034])
+    # Power = 0.5
+    plugin = OrographicSmoothingCoefficients(power=0.5)
+    expected = np.array([0.0, 0.707107, 1.0, 2.236068])
     result = plugin.unnormalised_smoothing_coefficients(gradient)
     assert_array_almost_equal(result[0, :], expected)
```
Comment (cgsandford) [inline improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py:174]: Can you make this more concise using `@pytest.parametrize`?

**Review by cgsandford — CHANGES_REQUESTED** [review]
One minor comment, plus: could you please regenerate the smoothing coefficients input to the recursive filter acceptance tests with the updated metadata, and check that the outputs are unchanged.

File: improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py (hunk 1 above)
Comment (bayliffe) [reply improver_tests/generate_ancillaries/test_OrographicSmoothingCoefficients.py:174]: Not without making it more opaque I fear. The expected values will have to get bunched up in the parameterize call, which I dislike. If you can envisage a neat way I'll implement it, but it's a pretty short test as it stands.

**Comment by bayliffe** [comment]
I've updated the recursive filter input files to include the new metadata in the acceptance test data update linked on the ticket.

**Review by cgsandford — APPROVED** [review]
Thanks for the changes - that's tidier.

**Review by gavinevans — APPROVED** [review]

**Pull request merged**